lms/djangoapps/discussion/rest_api/serializers.py

```diff
@@ -325,6 +336,13 @@ def get_unread_comment_count(self, obj):
             return obj["unread_comments_count"] + 1
         return obj["unread_comments_count"]
 
+    def get_preview_body(self, obj):
+        """
+        Returns a cleaned and truncated version of the thread's body to display in a
+        preview capacity.
+        """
+        return Truncator(strip_tags(obj["body"])).words(10, truncate=' …')
```

Comment:
The exact length of this is probably not as important, since the UI will truncate this again to fit the length.

Reply:
Can't we just add a field and truncate the body on the fronted? What are the benefits of doing it on the backend?
I guess, we would want to do it if a user doesn't have permissions to view it, or if we want to make very thin frontend which only displays the data, but I think neither are the case here.

Reply:
I guess the truncation here isn't that important, it's more important to strip the tags. That too can be done in the frontend though. So it's definitely up for debate whether this is a good addition.

Reply:
@xitij2000 Is the decision to keep this addition? Or do we have to follow the spec and can only remove it in the future?